src/comparator.js

```diff
@@ -0,0 +1,143 @@
+import csv from 'fast-csv'
+import {fetchConfig} from './config'
+import intersectionWith from 'lodash/intersectionWith'
+import differenceWith from 'lodash/differenceWith'
+
+type S3Path = {
+  Bucket: string,
+  Key: string
+}
+
+const s3 = new AWS.S3({ signatureVersion: 'v4' })
+
+export function handler (input, context, callback) {
+  compare().then((result) => callback(null, result)).catch((e) => {
+    console.log(e)
+    callback(e)
+  })
+}
+
+async function compare () {
+  let config = await fetchConfig()
+  const bucket = 'fulfilment-output-test'
+  const sfprefix = `${config.stage}/salesforce_output/`
+  const guprefix = `${config.stage}/fulfilment_output/`
+  const logprefix = `${config.stage}/comparator_output/`
+
+  console.log('Fetching existing files in S3: ', bucket)
+  console.log(sfprefix)
+  const sfresp = await s3.listObjectsV2({
+    Bucket: bucket,
+    Prefix: sfprefix
+  }).promise()
+  const sfkeys = sfresp.Contents.map(r => { return r.Key.slice(sfprefix.length) }).filter(notEmpty)
+  console.log('Found the following salesforce files', sfkeys)
+  console.log(guprefix)
+  const guresp = await s3.listObjectsV2({
+    Bucket: bucket,
+    Prefix: guprefix
+  }).promise()
+  const gukeys = guresp.Contents.map(r => { return r.Key.slice(guprefix.length) }).filter(notEmpty)
+  console.log('Found the following fulfilment files', gukeys)
+  console.log(logprefix)
+  const logresp = await s3.listObjectsV2({
+    Bucket: bucket,
+    Prefix: logprefix
+  }).promise()
+  const logkeys = logresp.Contents.map(r => { return r.Key.slice(guprefix.length) })
+  console.log('Found the following log files', logkeys)
+
+  const joint = intersectionWith(gukeys, sfkeys, (a:string, b:string) => {
+    return a.split('_').join('') === b.split('_').join('')
+  })
+  console.log('In both systems', joint)
+
+  const unchecked = differenceWith(joint, logkeys, (a, b) => {
+    return a === logFileNameFor(b)
+  })
+  console.log('remaining to check', unchecked)
+
+  if (unchecked.length === 0) {
+    return 'No files found to check.'
+  }
+
+  let sfPath = {Bucket: bucket, Key: `${sfprefix}${sfFilenameFor(unchecked[0])}`}
+  let guPath = {Bucket: bucket, Key: `${guprefix}${unchecked[0]}`}
+  let logPath = {Bucket: bucket, Key: `${logprefix}${logFileNameFor(unchecked[0])}`}
+
+  let logCache = []
+
+  let log = (entry) => {
+    console.log(entry)
+    logCache.push(entry)
+  }
+
+  let sfOutput = await fetchCSV(sfPath)
+  let guOutput = await fetchCSV(guPath)
+
+  Object.keys(sfOutput).forEach((id) => {
+    if (guOutput[id] === undefined || guOutput[id] == null) {
+      log(`Subscription ${id} found in Salesforce Output, but not fulfilment file.`)
+      return
+    }
+    let s = sfOutput[id]
+    let g = guOutput[id]
+    if (s.length !== g.length) {
+      log(`Differing numbers of fulfilments generated for ${id}. Salesforce: ${s.length} File: ${g.length}`)
+    }
+    delete guOutput[id]
+  })
+  Object.keys(guOutput).forEach((id) => {
```

Comment:
Maybe put a wee comment saying guOutput may be truncated by the previous code block.